openshift-kube-apiserver/admission/customresourcevalidation/customresourcevalidator.go

```diff
@@ -12,9 +12,9 @@ import (
 )
 
 type ObjectValidator interface {
-	ValidateCreate(obj runtime.Object) field.ErrorList
-	ValidateUpdate(obj runtime.Object, oldObj runtime.Object) field.ErrorList
-	ValidateStatusUpdate(obj runtime.Object, oldObj runtime.Object) field.ErrorList
+	ValidateCreate(ctx context.Context, obj runtime.Object) field.ErrorList
```

Comment:
Why do we need this change? I see just `TODO` and unused var being passed around?

Reply:
Ah, route validator now need a context. Lets add this as comment here or in the commit message

Reply:
Ti's the original discussion thread where ctx was suggested to be introduced in the ObjectValidator interface: https://github.com/openshift/kubernetes/pull/1659#discussion_r1288344050. To brief, it will essentially be required for certain router validation logic that is being planned to added as a part of `ExternalCertificateRef` feature.

However, for sake of simplicity I split the implementation into two chunks, this one being one of it where I'm bumping deps, adding the context as a no-op and adding feature gates in o/k-as. This is to ensure that we can add context here and merge these changes while later adding the router validation logic for feature gated `ExternalCertificateRef`. Please let me know if there are further clarifications regarding the scope, thanks!

Reply:
Sure, I will add a comment in code and commit msg :)